src/linear_algebra/determinant.lean

```diff
@@ -75,6 +76,24 @@ begin
   simp [det, card_eq_zero.mp h, perm_eq],
 end
 
+/-- If `n` has only one element, the determinant of an `n` by `n` matrix is just that element.
+Although `unique` implies `decidable_eq` and `fintype`, the instances might
+not be syntactically equal. Thus, we need to fill in the args explicitly. -/
+@[simp]
+lemma det_unique {n : Type*} [unique n] [decidable_eq n] [fintype n] (A : matrix n n R) :
+  det A = A (default n) (default n) :=
+by simp [det, univ_unique]
+
+lemma det_eq_elem_of_card_eq_one {A : matrix n n R} (h : fintype.card n = 1) (k : n) :
+  det A = A k k :=
+begin
+  have h1 : (univ : finset (perm n)) = {1},
+  { apply univ_eq_singleton_of_card_one (1 : perm n),
+    simp [card_univ, fintype.card_perm, h] },
+  have h2 := univ_eq_singleton_of_card_one k h,
+  simp [det, h1, h2],
+end
+
```

Comment:
This might make a nice standalone PR, if you want to get some parts merged sooner than others.

Reply:
I'll be lazy and patient...